src/clusterfuzz/_internal/base/tasks.py

```diff
@@ -412,6 +457,36 @@ def lease(self, _event=None):  # pylint: disable=arguments-differ
     track_task_end()
 
 
+class UTaskMainPubSubTask(PubSubTask):
+
+  def __init__(self, pubsub_message):
+    self._pubsub_message = pubsub_message
+    grandparent_class = super(PubSubTask, self)
+    grandparent_class.__init__(
+        self.attribute('command'),
+        self.attribute('argument'),
+        self.attribute('job'),
+        input_url=self.attribute('input_url'))
```

Comment:
This smells a bit weird.

Can `PubSubTask` instead initialize `input_url` unconditionally? That would be simplest. Given that this class is instantiated below only if `input_url` is not `None`, that seems like it would work. In fact, we might not need this class at all. AFAICT, no other code checks if a task is of this class to act in a special way.

If that would break things, then I suggest instead doing:

```suggestion
  def __init__(self, pubsub_message):
    super(UTaskMainPubSubTask, self).__init__(pubsub_message)
    self.input_url = self.attribute('input_url')
```